app/lib/katello/resources/cdn.rb

```diff
@@ -107,11 +112,16 @@ def http_downloader
           end
 
           # NOTE: This was added because some proxies dont support SSLv23 and do not handle TLS 1.2
-          # Valid values in ruby 1.9.3 are 'SSLv23' or 'TLSV1'
+          # Valid values in ruby 2.7.8 are :TLS1, :TLS1_1, :TLS1_2, :TLS1_3
           # Run the following command in rails console to figure out other
           # valid constants in other ruby versions
-          # "OpenSSL::SSL::SSLContext::METHODS"
-          net.ssl_version = @ssl_version
+          # "OpenSSL::SSL.constants.select { |sym| sym.to_s.include?('VERSION') }"
+          net.min_version = OpenSSL::SSL.const_get(
+            @cdn_min_tls_version.gsub('TLSv', 'TLS')
+                                .gsub('.', '_')
+                                .concat('_VERSION')
+                                .to_sym
```

Comment:
I'd consider pulling this out to a method to represent translation of our representation of TLS versions to the Ruby required ones and then it can be tested directly.

Reply:
And I would consider swapping these so that the list is based on the Ruby naming convention with a translation to something human readable. And, consider just having a map of values rather than a method that can break when naming convention breaks.

Reply:
Thanks for the suggestions. I updated it to use the map of values approach, which should be easy to maintain because the number of supported protocols is small.

I kept the human readable version strings as the keys and the Ruby naming convention as the values (we convert to the Ruby naming convention only once).

I also created a separate Array from the human readable version strings since those only are referenced in multiple files and it's easier than calling `SUPPORTED_TLS_VERSIONS_MAPPING.keys` each time.